### PR TITLE
WIP Factor-out auth-related logic

### PR DESCRIPTION
**Pull request by nl0**
Factor-out auth-related logic and wire-up the new auth backend. Log out on auth failure.

## TODO

- [x] tests for components:
  - [x] PassReset
  - [x] SignUp
  - [x] SignOut
  - [x] wrapper
- [x] move out reusable steps and other non-directly related stuff:
  - [x] CI and coverage stuff #668 
  - [x] extensible error #673 
  - [x] defer #670 
  - [x] validators #671 
  - [x] clipboard #669 
  - [x] context helpers and react updgrade #676
  - [x] redux tools #674 
  - [x] react tools #677
  - [x] mocks & testing tools #678
- [x] adjust for backend changes (status codes, etc)
- [ ] write a helper for api requests that catches 401s
- [x] manual testing with the latest api revision

### CODE REVIEW

File: catalog/.env.example [hunk 1]
```diff
@@ -26,21 +26,6 @@ REGISTRY_URL=http://flask:5000
...
-SIGN_OUT_URL=http://auth:5002/accounts/logout?next=%2F
-
-# github auth
-#USER_API=https://api.github.com/user
-#SIGN_OUT_URL=/
-
-# staging auth server
-#USER_API=https://stage-auth.quiltdata.com/accounts/api-root
-#SIGN_OUT_URL=https://stage-auth.quiltdata.com/accounts/logout?next=%2F
-
```
Comment (akarve) [inline catalog/.env.example]: I'm not clear why we dropped the above; are they no longer in static/config.js?

Comment (dimaryaz) [reply catalog/.env.example]: The auth server is gone, so there isn't anything to configure?

Comment (nl0) [reply catalog/.env.example]: > The auth server is gone, so there isn't anything to configure?

exactly

**Comment by nl0** [comment]
@akarve i've factored out the changes that are not directly related to auth into separate PRs. once they are merged, i'll create a clean PR with new auth UI against the flask auth branch.

**Comment by nl0** [comment]
closing in favor of #683

**Pull request closed without merging**